### ISSUE TITLE
Start swiper search with active region if there is one

### ISSUE THREAD

**Issue by JSpenced** [abo-abo/swiper#1884]
I was wondering if it's possible to have swiper start the search with the active region if there is one. I don't see a setting for this so maybe it isn't possible?

**Comment by mookid** [comment]
See #1868 and #1871

**Comment by JSpenced** [comment]
Thanks. For future reference it's done like this but not sure if this is the best way. The below keeps the region active which is good if you cancel the search but if you hit enter on the search ideally the active mark would be canceled:

```
(defun swiper-region ()
  "Use current region if active for swiper search"
  (interactive)
  (if (use-region-p)
      (swiper (format "%s" (buffer-substring (region-beginning) (region-end))))
    (swiper)))
```

**Comment by Alexander-Shukaev** [comment]
This better be solved once and for all in #1871.

**Comment by JSpenced** [comment]
I don't know lisp that well, so I didn't completely understand that PR. When the PR is merged into master, can you show how to utilize the PR grabbing the active region like above more effectively. @Alexander-Shukaev @mookid
